### PR TITLE
slight optimization for user/my{Entries} endpoints

### PR DESCRIPTION
**Pull request by Ldcabansay**
#3143 
Ideally would instead use a minimal entry object to populate the user/my{Entries} pages for a user, but would require some significant UI changes. Something to revisit later. 

For now, just doing a slight optimization by changing the endpoints to return only the relevant entry type for that endpoint (ex just Workflow, Tools, or Services). Loading myWorkflows for me was about 27% faster, and since I have very few Tools and Services, those pages now load near instantly.

### CODE REVIEW

File: dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/EntryDAO.java [hunk 1]
```diff
@@ -154,6 +154,10 @@ public T findPublishedById(long id) {
         return list(namedQuery("io.dockstore.webservice.core." + typeOfT.getSimpleName() + ".getEntryLiteByUserId").setParameter("userId", userId));
     }
 
+    public List<Entry> findMyEntries(long userId) {
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/EntryDAO.java]: I think this should return a `List<T>` intstead of `List<Entry>`. Then you shouldn't need to do the casts in UserResource.

Comment (Ldcabansay) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/EntryDAO.java]: complained apparently for services, so I'll just leave that one. Also found a few more places that could swap out to use the namedQueries

Comment (Ldcabansay) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/EntryDAO.java]: Complained for bioWorkflow too...

Comment (coverbeck) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/EntryDAO.java]: How did it complain? I remember talking about this with you, but at that time didn't realize the it was in this class with generics.

Comment (Ldcabansay) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/EntryDAO.java]: just that Service and BioWorkflow weren't of type Workflow, which is weird.

File: dockstore-webservice/src/main/resources/openapi3/openapi.yaml [hunk 2]
```diff
@@ -1417,6 +1500,8 @@ components:
             $ref: '#/components/schemas/FileFormat'
           type: array
           uniqueItems: true
+        parent:
+          $ref: '#/components/schemas/EntryObjectObject'
```
Comment (garyluu) [inline dockstore-webservice/src/main/resources/openapi3/openapi.yaml]: Where'd these changes come from?

Comment (denis-yuen) [reply dockstore-webservice/src/main/resources/openapi3/openapi.yaml]: I saw them start when merging the hotfix with develop. 
Might be some interaction between the new parentid column and some changes we did to openapi on develop

File: dockstore-webservice/src/main/resources/openapi3/openapi.yaml [hunk 3]
```diff
@@ -2902,7 +2902,7 @@ paths:
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/Aliasable'
+                $ref: '#/components/schemas/Collection'
```
Comment (garyluu) [inline dockstore-webservice/src/main/resources/openapi3/openapi.yaml]: Don't need to check this file in.

File: dockstore-integration-testing/src/test/java/io/dockstore/webservice/ServiceIT.java [hunk 4]
```diff
@@ -217,7 +218,14 @@ public void testGitHubAppEndpoints() throws Exception {
         // Test user endpoints
         UsersApi usersApi = new UsersApi(webClient);
         services = usersApi.userServices(service.getUsers().get(0).getId());
-        List<io.swagger.client.model.Workflow> workflows = usersApi.userWorkflows(service.getUsers().get(0).getId());
+        List<io.swagger.client.model.Workflow> workflows = new ArrayList<>();
+        try {
+            workflows.addAll(usersApi.userWorkflows(service.getUsers().get(0).getId()));
```
Comment (denis-yuen) [inline dockstore-integration-testing/src/test/java/io/dockstore/webservice/ServiceIT.java]: Not really sure I understand what is going on here. 
I would have expected services on line 220 to be filled with services, then workflows here to be filled with workflows. 
It seems odd to me to expect a 500 error?

**Pull request merged**